File: front/lib/utils/format_duration.dart
/// 125 => 2:05
///
String formatDuration(int durationInSeconds) {
  var seconds = durationInSeconds % 60;
  var minutes = durationInSeconds ~/ 60 % 60;
  var hours = durationInSeconds ~/ 3600;
  var formatedUpToMinutes =
      "${minutes.toString().padLeft(2, '0')}:${seconds.toString().padLeft(2, '0')}";
  if (hours > 0) {
    return "$hours:$formatedUpToMinutes";
  }
  return formatedUpToMinutes;
}

File: front-vue/utils/format-duration.ts
const format = (n: number) => n.toString().padStart(2, "0");
export const formatDuration = (timeInSeconds: number) => {
    const hours = Math.floor(timeInSeconds / 3600);
    const minutes = Math.floor(timeInSeconds / 60) % 60;
    const seconds = timeInSeconds % 60;

    if (hours) {
        return `${hours}:${format(minutes)}:${format(seconds)}`;
    }
    return `${format(minutes)}:${format(seconds)}`;
};
